src/core/geometry/qgsgeos.cpp

```diff
@@ -437,14 +437,14 @@ QgsAbstractGeometry *QgsGeos::combine( const QgsAbstractGeometry *geom, QString
 
 QgsAbstractGeometry *QgsGeos::combine( const QVector<QgsAbstractGeometry *> &geomList, QString *errorMsg, const QgsGeometryParameters &parameters ) const
 {
-  QVector< GEOSGeometry * > geosGeometries;
+  std::vector<geos::unique_ptr> geosGeometries;
```

Comment:
we need to use a std::vector rather a QVector, because QVector::resize() doesn't like unique_ptr (a least geos::unique_ptr) (probably requires a copy constuctor/assignment operator, whereas std::vector is fine with the move ones)

Reply:
Seems like a good idea in any case, we don't benefit from the implicit sharing here anyway.